packages/lesmis/src/components/YearSelector.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Select } from '@tupaia/ui-components';
+import { MIN_DATA_YEAR, ALL_DATES_VALUE } from '../constants';
+
+const getYearOptions = () => {
+  const years = [
+    {
+      label: 'All Data',
+      value: ALL_DATES_VALUE,
+    },
+  ];
+
+  const startYear = parseInt(MIN_DATA_YEAR);
+  let endYear = new Date().getFullYear();
+
+  while (startYear <= endYear) {
```

Comment:
```suggestion
  while (year >= startYear) {
```
Two suggestions:
- use `year` as variable name, because after the first iteration it's no longer the final year
- flip the order of the condition - this one's just personal preference, I always expect to see the variable that's changing on the left of the condition in a loop